### PR TITLE
Added Planar power controller

### PR DESCRIPTION
**Pull request by ppodhajski**

### CODE REVIEW

File: tide/core/types.h [hunk 1]
```diff
@@ -65,6 +65,16 @@ enum class TextureFormat
     yuv444
 };
 
+/**
+ * The power state of Planar displays.
+ */
+enum class screenState
```
Comment (rdumusc) [inline tide/core/types.h]: ScreenState

File: tide/master/LoggingUtility.cpp [hunk 2]
```diff
@@ -115,6 +115,22 @@ void LoggingUtility::_incrementWindowCount()
     ++_windowCounter;
 }
 
+void LoggingUtility::powerStateChanged(screenState state)
```
Comment (rdumusc) [inline tide/master/LoggingUtility.cpp]: const ScreenState state

File: tide/master/MasterApplication.cpp [hunk 3]
```diff
@@ -406,6 +411,13 @@ void MasterApplication::_initTouchListener()
     connect(_touchListener.get(), &MultitouchListener::touchPointAdded,
             [this, getWallPos](const int id, const QPointF normalizedPos) {
                 _markers->addMarker(id, getWallPos(normalizedPos));
+#if TIDE_ENABLE_PLANAR_CONTROLLER
+                if (_planarController->getState() == screenState::OFF)
+                    if (!_planarController->powerOn())
+                        put_flog(LOG_INFO,
+                                 "Could not power on the screens"
```
Comment (rdumusc) [inline tide/master/MasterApplication.cpp]: "Could not power on the screens by " << and don't forget to add a space before "

File: tide/master/rest/RestLogger.cpp [hunk 4]
```diff
@@ -44,6 +44,18 @@
 
 #include <QJsonObject>
 
+namespace
+{
+QString stateToString(screenState state)
+{
+    if (state == screenState::ON)
+        return "ON";
+    else if (state == screenState::OFF)
```
Comment (rdumusc) [inline tide/master/rest/RestLogger.cpp]: else not needed - should be a switch normally

File: tide/master/PlanarController.h [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    void checkPowerState();
+
+    /** Power off Planar displays */
+    bool powerOff();
+
+    /** Power on Planar displays */
+    bool powerOn();
+
+signals:
+    void powerStateChanged(screenState state);
```
Comment (rdumusc) [inline tide/master/PlanarController.h]: missing doxygen

File: tide/master/PlanarController.h [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+#ifndef PLANARCONTROLLER_H
+#define PLANARCONTROLLER_H
+
+#include "types.h"
+
+#include <QObject>
+#include <QSerialPort>
+#include <QTimer>
+
+class PlanarController : public QObject
```
Comment (rdumusc) [inline tide/master/PlanarController.h]: Missing class doxygen

File: tide/master/PlanarController.cpp [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+bool PlanarController::powerOn()
+{
+    if (_serial.isOpen())
+    {
+        _serial.write("OPA1DISPLAY.POWER=ON\r");
+        return _serial.waitForBytesWritten(serialTimeout);
+    }
+    else
```
Comment (rdumusc) [inline tide/master/PlanarController.cpp]: else not needed, there is a return above. You could also format it with an early out:
if(!_serial.isOpen())
{
put_flog(...);
return false;
}
_serial.write("OPA1DISPLAY.POWER=ON\r");
return _serial.waitForBytesWritten(serialTimeout);

But is the isOpen() test still needed now that you added a throw in the constructor?

Comment (ppodhajski) [inline tide/master/PlanarController.cpp]: Not at all, I wanted to discuss which one we choose.

File: tide/master/MasterApplication.cpp [hunk 8]
```diff
@@ -503,7 +504,12 @@ void MasterApplication::_initRestInterface()
                 _logger.get(), &LoggingUtility::powerStateChanged);
 
         connect(_restInterface.get(), &RestInterface::powerOff, [this]() {
-            if (!_planarController->powerOff())
+            if (_planarController->powerOff())
+            {
+                DisplayGroupController controller(*_displayGroup.get());
```
Comment (rdumusc) [inline tide/master/MasterApplication.cpp]: no .get() on smart ptrs!! also can be on a single line: DisplayGroupController(*_displayGroup).hidePanels();

**Pull request merged**